Paintroid/src/main/java/org/catrobat/paintroid/iotasks/CreateFile.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Paintroid: An image manipulation application for Android.
+ * Copyright (C) 2010-2021 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.paintroid.iotasks
+
+import android.app.Activity
+import android.util.Log
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.catrobat.paintroid.FileIO
+import java.io.File
+import java.lang.NullPointerException
+import java.lang.ref.WeakReference
+
+class CreateFile(
+    callback: CreateFileCallback,
+    private val requestCode: Int,
+    private val filename: String?,
+    private val scopeIO: CoroutineScope
+) {
+    private val callbackRef: WeakReference<CreateFileCallback> = WeakReference(callback)
```

Comment:
Hmm. Good.
But I think there is smth odd in other classes.
Could you please check the usages and hope there isn't anything like:
Activity implements CreateFileCallback {
  fileActivity = this
  ...
}

Reply:
There's not anything like:
Activity implements CreateFileCallback {
fileActivity = this
...
}
But the MainActivityPresenter overrides the fileActivity var in the constructor. I think that shouldn't be a problem? 🤔

Reply:
and the context is passed in the MainActivity.